references/cli/adopt.go

```diff
@@ -0,0 +1,453 @@
+		see the default implementation of adoption rules.
+	`))
+	adoptExample = templates.Examples(i18n.T(`
+		# Native Resources Adoption
+
+		## Adopt resources into new application
+		vela adopt deployment/my-app configmap/my-app
+
+		## Adopt resources into new application with specified app name
+		vela adopt deployment/my-deploy configmap/my-config --app-name my-app
+
+		## Adopt resources into new application in specified namespace
+		vela adopt deployment/my-app configmap/my-app -n demo
+
+		## Adopt resources into new application across multiple namespace
+		vela adopt deployment/ns-1/my-app configmap/ns-2/my-app
+
+		## Adopt resources into new application with take-over mode
+		vela adopt deployment/my-app configmap/my-app --mode take-over
+
+		## Adopt resources into new application and apply it into cluster
+		vela adopt deployment/my-app configmap/my-app --apply
+
+		-----------------------------------------------------------
+
+		# Helm Chart Adoption
+
+		## Adopt resources in a deployed helm chart
+		vela adopt my-chart -n my-namespace --type helm
+		
+		## Adopt resources in a deployed helm chart with take-over mode
+		vela adopt my-chart --type helm --mode take-over
+
+		## Adopt resources in a deployed helm chart in an application and apply it into cluster
+		vela adopt my-chart --type helm --apply
+
+		## Adopt resources in a deployed helm chart in an application, apply it into cluster, and recycle the old helm release after the adoption application successfully runs
+		vela adopt my-chart --type helm --apply --recycle
+
+		-----------------------------------------------------------
+
+		## Customize your adoption rules
+		vela adopt my-chart -n my-namespace --type helm --adopt-template my-rules.cue
+	`))
+)
+
+// NewAdoptCommand command for adopt resources into KubeVela Application
+func NewAdoptCommand(f velacmd.Factory, streams util.IOStreams) *cobra.Command {
+	o := &AdoptOptions{
+		Type:      adoptTypeNative,
+		Mode:      adoptModeReadOnly,
+		IOStreams: streams,
+	}
+	cmd := &cobra.Command{
+		Use:     "adopt",
+		Short:   i18n.T("Adopt resources into new application"),
+		Long:    adoptLong,
+		Example: adoptExample,
+		Annotations: map[string]string{
+			types.TagCommandType: types.TypeCD,
+		},
+		Args: cobra.MinimumNArgs(1),
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(o.Complete(f, cmd, args))
+			cmdutil.CheckErr(o.Validate())
+			cmdutil.CheckErr(o.Run(f, cmd))
+		},
+	}
+	cmd.Flags().StringVarP(&o.Type, "type", "t", o.Type, fmt.Sprintf("The type of adoption. Available values: [%s]", strings.Join(adoptTypes, ", ")))
+	cmd.Flags().StringVarP(&o.Mode, "mode", "m", o.Mode, fmt.Sprintf("The mode of adoption. Available values: [%s]", strings.Join(adoptModes, ", ")))
+	cmd.Flags().StringVarP(&o.AppName, "app-name", "", o.AppName, "The name of application for adoption. If empty for helm type adoption, it will inherit the helm chart's name.")
```

Comment:
why don't just use `name`?

Reply:
because `name` could be ambiguous, like `resource name` / `helm chart name` / `app name`, ...